test/test_podman_baseline.sh

```diff
@@ -90,6 +90,52 @@ podman run $image java 2>&1 || echo $?
 ########
 podman rm --all
 
+########
+# pull and run many containers in parallel, test locks ..etc.
+########
+podman rmi docker.io/library/busybox:latest > /dev/null || :
+for i in `seq 10`
+do ( podman run -d --name b$i docker.io/library/busybox:latest busybox httpd -f -p 80 )&
+done
+echo -e "\nwaiting for creation...\n"
+wait
+echo -e "\ndone\n"
+# assert we have 10 running containers
+count=$( podman ps -q  | wc -l )
+[ "x$count" == "x10" ] || { echo "was expecting 10 running containers"; exit -1; }
+
+for i in `seq 10`; do ( podman stop -t=1 b$i; podman rm b$i )& done
+echo -e "\nwaiting for deletion...\n"
+wait
+echo -e "\ndone\n"
+# assert we have 0 running containers
+count=$( podman ps -q  | wc -l )
+[ "x$count" == "x0" ] || { echo "was expecting 0 running containers"; exit -1; }
+
+
+########
+# run many containers in parallel for an existing image, test locks ..etc.
+########
+podman pull docker.io/library/busybox:latest > /dev/null || :
+for i in `seq 10`
+do ( podman run -d --name c$i docker.io/library/busybox:latest busybox httpd -f -p 80 )&
+done
+echo -e "\nwaiting for creation...\n"
+wait
+echo -e "\ndone\n"
+# assert we have 10 running containers
+count=$( podman ps -q  | wc -l )
+[ "x$count" == "x10" ] || { echo "was expecting 10 running containers"; exit -1; }
+
```

Comment:
Well this is interesting.  The following removal fails for me with:

```
Error: unable to find container c1: more than one result for ID or name c1: container already exists
Error: unable to find container c1: more than one result for ID or name c1: container already exists
```
Unless I put a several second sleep here.  I then tried to changed lines 130 -> 133 with `podman rmi -a -f`, that seems to work, but takes a very long time, 30+ seconds of no feedback.

Adding a '-f' to the rm command on line 130 seems to make the error happen less frequently, but it's still pops from time to time.

@mheon, @vrothberg thoughts?  Is this pulling up a lock issue?

Reply:
Also fwiw, I stopped the script in a few places, and looked at the created containers and we'd no duplicates, so I think at least the error could use some massaging.

@muayyad-alsadi these two comments I don't suspect are an issue with your script, rather they may have found something interesting in Podman.

Reply:
@TomSweeneyRedHat oh, this is an interesting one - 1 in 256 chance that we get a container ID that starts with C1, in which case we have a container c1xxxxxx... and a container with the name c1 - and `rm` can't tell which one we wanted to remove.

Need to check Docker for expected behavior here.

Reply:
yes, this is the exact case those tests are made to catch. and I guess #2553 is also related.

it happened once, but with yesterday version of `podman` it did not report error.

Reply:
> container ID that starts with C1, in which case we have a container c1xxxxxx...

I can change it to c_1 to fix that if you like.

Reply:
@muayyad-alsadi I think this might well be a bug - Docker handles this by deleting the container named `c1` even if there is an ID match as well.

Reply:
I think I'm going to submit a bugfix PR for this.

Reply:
Opened #2563 to fix our handling of this